app/javascript/components/reconfigure-vm-form/helpers/network.js

```diff
@@ -0,0 +1,301 @@
+import { TYPES, formsData } from './general';
+
+/** Function which executes when the cancel edit button gets clicked */
+export const cancelEditNetwork = (row, data, setData, roles) => {
+  const index = row.id.substring(7);
+  const networkRows = data.dataTable.networkAdapters;
+
+  if (!roles.isVmwareCloud) {
+    networkRows[index].vlan = networkRows[index].old_vlan;
+  }
+
+  if (roles.isVmwareCloud) {
+    networkRows[index].network = networkRows[index].old_network;
+  }
+  const submitNetworks = data.submitParams.networks.edit.filter((network) => network.id !== row.id);
+  const editedNetworkIds = data.submitParams.networkIds.edit.filter((id) => id !== row.id);
+
+  data.dataTable.networkAdapters = networkRows;
+  data.changeCounter -= 1;
+
+  setData({
+    ...data,
+    submitParams: {
+      ...data.submitParams,
+      networks: {
+        ...data.submitParams.networks,
+        edit: submitNetworks,
+      },
+      networkIds: {
+        ...data.submitParams.networkIds,
+        edit: editedNetworkIds,
+      },
+    },
+  });
+};
+
+/** Function which executes when the delete button gets clicked */
+export const deleteNetwork = (row, data, setData, roles) => {
+  let dataTableData = data.dataTable.networkAdapters || [];
+  let submitNetworks = data.submitParams.networks.add || [];
+  let addNetworkIds = data.submitParams.networkIds.add || [];
+  const deleteNetworksData = data.submitParams.networks.delete || [];
+  const deletedNetworkIds = data.submitParams.networkIds.delete || [];
+
+  if (data.submitParams.networkIds.add.includes(row.id)) {
+    // remove from the newly added networks
+    // eslint-disable-next-line no-alert
+    if (window.confirm(__('Are you sure you want to delete this entry?'))) {
```

Comment:
maybe we can change the confirmation to carbon in a different PR

Reply:
Sure.